docs/reference/glossary.md

```diff
@@ -181,22 +187,60 @@ labeled using [tags](#tag).
 Here is an example of the shared [nginx repository](https://hub.docker.com/_/nginx/)
 and its [tags](https://hub.docker.com/r/library/nginx/tags/)
 
-## Swarm
+## routing mesh
+
+The swarm mode ingress [routing mesh](../swarm/ingress.md) lets you expose ports
+for services externally to a swarm.
+
+## service
+
+A [service](../swarm/how-swarm-mode-works/services.md) is the central structure
+of the swarm mode system and the primary root of user interaction with the
+swarm. To deploy an application image when Docker Engine is in swarm mode, you
```

Comment:
By reading this first sentence you'll still don't know what a service is. What about starting the explanation with something users already know, like containers?

A service is a collection of containers running the exact same task. When you scale a service, new containers are started, and requests are automatically load-balanced across them.